### PR TITLE
clippy: fix warnings in components/shared

### PR DESCRIPTION
**Pull request by eerii**
Split from #31514, fixes some clippy warnings in `components/shared`.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #31500
- [x] These changes do not require tests because they do not modify functionality, they only fix lint errors.

### CODE REVIEW

**Review by mrobinson — CHANGES_REQUESTED** [review]
Nice work. Just a couple small comments below:

File: components/shared/msg/constellation_msg.rs [hunk 1]
```diff
@@ -209,16 +211,18 @@ namespace_id! {PipelineId, PipelineIndex, "Pipeline"}
...
-    pub fn new() -> PipelineId {
+impl Default for PipelineId {
+    fn default() -> Self {
         PIPELINE_NAMESPACE.with(|tls| {
             let mut namespace = tls.get().expect("No namespace set for this thread!");
             let new_pipeline_id = namespace.next_pipeline_id();
             tls.set(Some(namespace));
             new_pipeline_id
         })
     }
```
[the remark this answers is not in the document]
Comment (eerii) [reply components/shared/msg/constellation_msg.rs:222]: I think that makes a ton of sense. I'll revert the changes for this types that end in `..Id`.

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
This needs a rebase, but otherwise looks great. Thanks!

**Comment by eerii** [comment]
Thanks for the review! It should be rebased to main now

**Pull request merged**